state-synchronizer/src/executor_proxy.rs

```diff
@@ -205,19 +200,20 @@ impl ExecutorProxyTrait for ExecutorProxy {
         }
 
         // calculate deltas
-        let new_configs = self.fetch_all_configs().await;
+        let new_configs = self.fetch_all_configs().await?;
         let changed_configs = ON_CHAIN_CONFIG_REGISTRY
             .iter()
             .cloned()
             .filter(|id| {
-                // the byte array is `None` on storage read failure
-                // We should still publish reconfig notification for failed DB read, to indicate that a
-                // reconfig event did happen, regardless of DB failure
-                // So, we only do not publish a config if its previous and new values are guaranteed
-                // to be the same
-                let new_cfg = new_configs.get(id).cloned().flatten();
-                let cfg = self.on_chain_configs.get(id).cloned().flatten();
-                !(new_cfg == cfg && new_cfg.is_some())
+                // Only publish if config changed
+                let new_cfg = new_configs
+                    .get(id)
+                    .expect("missing on-chain config value from DB query");
```

Comment:
wonder how do we represent uninitialized config here, or the bigger story how we bootstrap a config not in genesis? @runtian-zhou sth like this
1. develop the move module
2. publish move module
3. initialize the config value
4. register the config id + develop code depending on the config value
This way we guarantee any registered config id has a value.